src/main/java/co/cask/bigquery/BigQuerySource.java

```diff
@@ -249,20 +250,27 @@ public Schema getSchema(Request request) throws Exception {
         } else if (value == LegacySQLTypeName.BYTES) {
           cType = Schema.Type.BYTES;
         } else if (value == LegacySQLTypeName.TIME) {
-          cType = Schema.Type.STRING;
+          lType = Schema.LogicalType.TIME_MICROS;
         } else if (value == LegacySQLTypeName.DATE) {
-          cType = Schema.Type.STRING;
-        } else if (value == LegacySQLTypeName.DATETIME) {
-          cType = Schema.Type.STRING;
+          lType = Schema.LogicalType.DATE;
+        } else if (value == LegacySQLTypeName.TIMESTAMP) {
+          lType = Schema.LogicalType.TIMESTAMP_MICROS;
         } else if (value == LegacySQLTypeName.RECORD) {
           throw new Exception("Nested records not supported yet.");
         }
 
         if (field.getMode() == null || field.getMode() == Field.Mode.NULLABLE) {
           Schema fieldType = Schema.nullableOf(Schema.of(cType));
+          if (lType != null) {
+            fieldType = Schema.nullableOf(Schema.of(lType));
+          }
           fields.add(Schema.Field.of(field.getName(), fieldType));
         } else if (field.getMode() == Field.Mode.REQUIRED) {
```

Comment:
this won't be needed if we use a Schema variable instead of a Schema.Type above.